include/seqan3/io/structure_file/format_vienna.hpp

```diff
@@ -0,0 +1,425 @@
+ * containing one or more sequences. Each sequence must appear as a single line in the file.
+ * A sequence may be preceded by a special line starting with the '>' character followed by a sequence name
+ * (like FastA). After each sequence line there is usually a line containing
+ * secondary structure, using brackets to denote interacting nucleotides or amino acids, and dots for unpaired sites.
+ * The length of the struture must equal the length of the sequence.
+ * Optionally, the structure may be followed by a space character and the minimum free energy value enclosed
+ * in parentheses (). Note that there cannot be energy without structure.
+ *
+ * The Vienna format is the output format of _RNAfold_. Furthermore, it is designed to be compatible with the
+ * input format of the ViennaRNA package (if structure and energy are omitted).
+ * See https://www.tbi.univie.ac.at/RNA/tutorial/#sec2_7 for details.
+ *
+ * ### Fields
+ *
+ * The Vienna format provides the fields seqan3::field::SEQ, seqan3::field::ID, seqan3::field::BPP (read only),
+ * seqan3::field::STRUCTURE, seqan3::field::STRUCTURED_SEQ and seqan3::field::ENERGY.\n\n
+ * If you select seqan3::field::STRUCTURED_SEQ you must not select seqan3::field::SEQ or seqan3::field::STRUCTURE.\n
+ * Either the field seqan3::field::SEQ or the field seqan3::field::STRUCTURED_SEQ is required when writing.\n
+ * The field seqan3::field::BPP is ignored when writing, but a structure string can be converted to bpp when reading.
+ *
+ * ### Implementation notes
+ *
+ * In the unlikely case that you want to read a Vienna file, which does not contain structure lines, you have to
+ * assign false to `seqan3::structure_file_input_options::file_has_structure`.
+ * ```cpp
+ *    structure_file_input_options<rna4, false> options;
+ *    options.file_has_structure = false;
+ * ```
+ *
+ * When reading the ID-line the identifier (`>`) and any blank characters before the actual ID are
+ * stripped. Each field is read/written as a single line (except ENERGY, which goes right after the structure).
+ * Numbers and spaces within the sequence are simply ignored, but not within the structure.
+ */
+class structure_file_format_vienna
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \brief Rule of five explicitly defaulted.
+     * \{
+     */
+    structure_file_format_vienna() = default;
+    structure_file_format_vienna(structure_file_format_vienna const &) = delete;
+    structure_file_format_vienna & operator=(structure_file_format_vienna const &) = delete;
+    structure_file_format_vienna(structure_file_format_vienna &&) = default;
+    structure_file_format_vienna & operator=(structure_file_format_vienna &&) = default;
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "dbn" }
```

Comment:
in SeqAn2 we have `.dbv`, too, is that not used anymore?

Reply:
It is not used anymore, but I added `.fa` and `.fasta` instead.